pkg/authn/k8schain/k8schain.go

```diff
@@ -46,11 +46,11 @@ func New(ctx context.Context, client kubernetes.Interface, opt Options) (authn.K
 	}
 
 	return authn.NewMultiKeychain(
+		k8s,
 		authn.DefaultKeychain,
```

Comment:
```suggestion
		authn.DefaultKeychain,
		k8s,
```

I think we may still want the default keychain first, and it should at least be much quicker than the others, barring any cred helper shenanigans.

WDYT?

Reply:
I'm ok with that .

Just be aware that if you're testing k8schain (pulling from creds from secrets) and you're hitting a real cluster you'll want to `docker logout` beforehand :)